tests/pipelines/stable_diffusion_xl/test_stable_diffusion_xl_adapter.py

```diff
@@ -560,3 +564,64 @@ def test_inference_batch_single_identical(
 
         if test_mean_pixel_difference:
             assert_mean_pixel_difference(output_batch[0][0], output[0][0])
+
+
+@slow
+@require_torch_gpu
+class AdapterSDXLPipelineSlowTests(unittest.TestCase):
+    def tearDown(self):
+        super().tearDown()
+        gc.collect()
+        torch.cuda.empty_cache()
+
+    def test_canny(self):
+        adapter = T2IAdapter.from_pretrained(
+            "TencentARC/t2i-adapter-lineart-sdxl-1.0", torch_dtype=torch.float16
+        ).to("cpu")
+        pipe = StableDiffusionXLAdapterPipeline.from_pretrained(
+        'stabilityai/stable-diffusion-xl-base-1.0', adapter=adapter, torch_dtype=torch.float16, variant="fp16", 
+        )
+        pipe.load_lora_weights("CiroN2022/toy-face", weight_name="toy_face_sdxl.safetensors")
+        pipe.enable_sequential_cpu_offload()
+        pipe.set_progress_bar_config(disable=None)
+
+        generator = torch.Generator(device="cpu").manual_seed(0)
+        prompt = "toy"
+        image = load_image(
+                    "https://huggingface.co/datasets/hf-internal-testing/diffusers-images/resolve/main/t2i_adapter/toy_canny.png"
+        )
+
+        images = pipe(prompt, image=image, generator=generator, output_type="np", num_inference_steps=3).images
+
+        assert images[0].shape == (768, 512, 3)
+
+        original_image = images[0, -3:, -3:, -1].flatten()
+        expected_image = np.array([0.5226907,  0.5272584,  0.52255356, 0.50762856, 0.50497484, 0.50656825, 0.5065469,  0.51809096, 0.48973307])
+        assert np.allclose(original_image, expected_image, atol=1e-04)
+
+
+    def test_canny_lora(self):
+        adapter = T2IAdapter.from_pretrained(
+            "TencentARC/t2i-adapter-lineart-sdxl-1.0", torch_dtype=torch.float16
+        ).to("cpu")
+        pipe = StableDiffusionXLAdapterPipeline.from_pretrained(
+        'stabilityai/stable-diffusion-xl-base-1.0', adapter=adapter, torch_dtype=torch.float16, variant="fp16", 
+        )
+        pipe.load_lora_weights("CiroN2022/toy-face", weight_name="toy_face_sdxl.safetensors")
+        pipe.enable_sequential_cpu_offload()
+        pipe.set_progress_bar_config(disable=None)
+
+        generator = torch.Generator(device="cpu").manual_seed(0)
+        prompt = "toy"
+        image = load_image(
+                    "https://huggingface.co/datasets/hf-internal-testing/diffusers-images/resolve/main/t2i_adapter/toy_canny.png"
+        )
+
+        images = pipe(prompt, image=image, generator=generator, output_type="np", num_inference_steps=3).images
+
+        assert images[0].shape == (768, 512, 3)
+
+        original_image = images[0, -3:, -3:, -1].flatten()
+        expected_image = np.array([0.50346327, 0.50708383, 0.50719553, 0.5135172,  0.5155377,  0.5066059, 0.49680984, 0.5005894,  0.48509413])
+        assert np.allclose(original_image, expected_image, atol=1e-04)
```

Comment:
Use `numpy_cosine_similarity_distance` on the arrays

```suggestion
        assert numpy_cosine_similarity_distance(original_image, expected_image) < 1e-4
```